### DISCUSSION TITLE
[油猴脚本]BE夜间模式样式修复

### DISCUSSION THREAD

**Discussion by loktindyi** [the1812/Bilibili-Evolved#3525] [分享 / Share]
不知道为什么现在这个项目更新得很慢，夜间模式总是来不及适配，各种白块
也不知道有没有人写这种脚本，总是我还是写了
效果图我懒得放了，现阶段也就是修复了首页和视频页的样式
未来还有计划：
 - [x] https://github.com/the1812/Bilibili-Evolved/issues/3506
 - [ ] https://github.com/the1812/Bilibili-Evolved/issues/3483
 - [ ] https://github.com/the1812/Bilibili-Evolved/discussions/3520
 - [ ] https://github.com/the1812/Bilibili-Evolved/discussions/3508
 - [ ] 有界面的“正则匹配=>注入样式”配置

脚本链接：https://greasyfork.org/zh-CN/scripts/448495
脚本默始终注入夜间样式，如果你有开启夜间计划时段，你可以：
 1. 打开Bilibili-Evolved(后简称BE)设置，点击`夜间模式计划时段`，此时你能看见你设置的计划时段了
 2. 点击油猴脚本菜单，点击`获取夜间计划`，这样就OK啦

可以在油猴脚本菜单点击`加入星凰`来提出问题与建议
懂正则和样式的，可以直接修改脚本里`regMatchStyle`变量来修补更多页面样式，但这只能是暂时的

**Comment by the1812** [comment]
> 不知道为什么现在这个项目更新得很慢，夜间模式总是来不及适配，各种白块

因为少了个人啊
<img src="https://user-images.githubusercontent.com/26504152/181034742-32be48f0-7fad-4606-b948-a34884f30d59.png" width="40">

![image](https://user-images.githubusercontent.com/26504152/181033748-1f6863ad-f8a0-41b7-a33a-2c0b1dc2ea41.png)

**Reply by loktindyi** [reply]
这可太难受了，这个大项目我看不懂，不然pr是最好的选择，夜间模式是哪一块，我可能可以试试，也许

**Reply by the1812** [reply]
可能和你的开发风格差别比较大, 文档在 https://github.com/the1812/Bilibili-Evolved/blob/preview/CONTRIBUTING.md
源码在 https://github.com/the1812/Bilibili-Evolved/blob/preview-fixes/registry/lib/components/style/dark-mode/

-------

另外, 你的脚本如果想控制在仅开启了夜间模式时生效, 可以用 `body.dark` 来检测
```css
/* 只在夜间模式开启时生效 */
body.dark a {
   color: #eee !important;
}

**Comment by Geeyun-JY3** [comment]
- 额 #3506 是因为这属于新版播放页，说了暂时不适配：#2615、[v2.2.0](https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.2.0)，没关而已
- #3483 这种特殊直播间一直没怎么适配的，破站会塞各种乱七八糟的元素进去，而且时效性还强
- #3520、#3508 其实并不算夜间模式的问题了

没适配的可以看这里：#641、#656、#731、#877、#1143、#1289、#1692、#2136、#2313、#2934、#2945、#3193

**Reply by loktindyi** [reply]
#3506 正是因为适配慢，我才做的这个脚本，因为我自己也要用
我没有提到过 #2615 吧
#3483 我实际没有适配特殊直播间的想法，引用这个issue是错误的，实际我要解决通用直播间的样式问题（现在也还有白块问题）
#3520 #3508 这当然的确不是夜间样式问题，只是我也对这个功能感兴趣，有意向写出这样的功能

你的issue很多，我会尝试先行适配，小油猴脚本的好处是更新比较快，但登录页这种少用的优先级一定会比较低

**Reply by Geeyun-JY3** [reply]
引用 #2615 是因为这就是新版视频播放页的 issue，#3506 算是其中一部分，先前有多个重复 issue 都关了，这个没关而已

我列出这些 issue 就是给你看一下 Bilibili Evolved 夜间模式现状，不求适配😂

**Reply by loktindyi** [reply]
适配还是要做的，至少常用的页面我会尽快做完，因为我自己也在用，看着不舒服
你的issue算是给了我一个TODO列表，我可以很快找到方向，一步一步来
